src/exchange/group/accounts/group-accounts.controller.js

```diff
@@ -74,51 +74,51 @@ angular.module("Module.exchange.controllers")
-                .Exchange
-                .getAccountsByGroup(this.$routerParams.organization, this.$routerParams.productId, this.selectedGroup.mailingListAddress, count, offset, this.search != null ? this.search.value : "")
+            return this.Exchange
+                .retrievingAccountsByGroup(this.$routerParams.organization, this.$routerParams.productId, this.selectedGroup.mailingListAddress, count, offset, this.search != null ? this.search.value : "")
                 .then((accounts) => {
                     this.accountsListBuffer = accounts;
                     this.accountsList = angular.copy(accounts);
                 })
                 .catch((failure) => {
-                    this.services.messaging.writeError(this.services.translator.tr("exchange_tab_ACCOUNTS_error_message"), failure);
+                    this.messaging.writeError(this.translator.tr("exchange_tab_ACCOUNTS_error_message"), failure);
                 })
                 .finally(() => {
                     this.loading = false;
-                    this.services.$scope.$broadcast("paginationServerSide.loadPage", 1, "accountsByGroupTable");
+                    this.$scope.$broadcast("paginationServerSide.loadPage", 1, "accountsByGroupTable");
                 });
         }
 
-        updateAccounts () {
-            this.services.messaging.writeSuccess(this.services.translator.tr("exchange_dashboard_action_doing"));
+        updatingAccounts () {
+            this.messaging.writeSuccess(this.translator.tr("exchange_dashboard_action_doing"));
             this.saveSelection();
 
-            this.services
-                .Exchange
-                .updateGroups(this.$routerParams.organization, this.$routerParams.productId, this.selectedGroup.mailingListAddress, this.model)
+            return this.Exchange
+                .updatingGroups(this.$routerParams.organization, this.$routerParams.productId, this.selectedGroup.mailingListAddress, this.model)
                 .then((data) => {
                     const addGroupMessages = {
-                        OK: this.services.translator.tr("exchange_GROUPS_settings_success_message", this.selectedGroup.mailingListDisplayName),
-                        PARTIAL: this.services.translator.tr("exchange_GROUPS_settings_partial_message", this.selectedGroup.mailingListDisplayName),
-                        ERROR: this.services.translator.tr("exchange_GROUPS_settings_error_message", this.selectedGroup.mailingListDisplayName)
+                        OK: this.translator.tr("exchange_GROUPS_settings_success_message", this.selectedGroup.mailingListDisplayName),
+                        PARTIAL: this.translator.tr("exchange_GROUPS_settings_partial_message", this.selectedGroup.mailingListDisplayName),
+                        ERROR: this.translator.tr("exchange_GROUPS_settings_error_message", this.selectedGroup.mailingListDisplayName)
                     };
 
                     if (data == null) {
```

Comment:
What do you think about : https://lodash.com/docs/3.10.1#isEmpty ? More cases will be tested as `{}` `[]` `""` etc :)